db.go

```diff
@@ -69,6 +69,14 @@ func (db *DB) Insert(collectionName string, docs ...*Document) error {
 	return db.engine.Insert(collectionName, insertDocs...)
 }
 
+// Save or update a document
+func (db *DB) Save(collectionName string, doc *Document) error {
+	if !doc.Has(objectIdField) {
+		return db.Insert(collectionName, doc)
+	}
+	return db.engine.Insert(collectionName, doc)
```

Comment:
In the last commits I added a `ReplaceById()` function. It is better to use it at line 77, because in the case the supplied document id doesn't exist, it will return an error.